nvflare/private/fed/server/client_manager.py

```diff
@@ -141,14 +141,17 @@ def authenticated_client(self, request, context) -> Optional[Client]:
                 )
                 return None
 
-            for token, client in self.clients.items():
-                if client.name == client_name:
-                    with self.lock:
-                        self.clients.pop(token)
+            with self.lock:
+                clients_to_be_moved = []
+                for token, client in self.clients.items():
+                    if client.name == client_name:
+                        clients_to_be_moved.append(token)
                         self.logger.info(
                             f"Client: {client_name} already registered. Re-login the client with a new token."
                         )
                         break
+                for item in clients_to_be_moved:
+                    self.clients.pop(item)
 
```

Comment:
Since only removing one client we could just do one item instead of clients_to_be_moved.

Reply:
we could remove all the client with the same client_name items. Most likely there's only one.